### PR TITLE
soft_504: change front adc read scale

### PR DESCRIPTION
**Pull request by JarvisWeng**

### CODE REVIEW

File: projects/power_distribution/src/pd_fan_ctrl.c [hunk 1]
```diff
@@ -56,7 +57,18 @@ static void prv_front_temp_to_fan_percent(uint16_t v_measured, uint8_t *fan_spee
     *fan_speed = 100;
     return;
   }
-  double ratio = s_fan_storage.ref_reading / 100.0;
+  if (v_measured == 0) {
```
Comment (AveryChiu64) [inline projects/power_distribution/src/pd_fan_ctrl.c:60]: huh, do we ever divide `v_measured` by 0?

File: projects/power_distribution/src/pd_fan_ctrl.c [hunk 2]
```diff
@@ -56,7 +57,18 @@ static void prv_front_temp_to_fan_percent(uint16_t v_measured, uint8_t *fan_spee
...
+    // just to be safe to avoid divide-by-zero errors
+    *fan_speed = 0;
+    return;
+  }
+
+  if (v_measured > (FRONT_FAN_CTRL_MAX_VALUE_MV + ADC_EPSILON_MV)) {
```
Comment (AveryChiu64) [inline projects/power_distribution/src/pd_fan_ctrl.c:66]: Maybe we could check for both these conditions in line 60, eg
`if` (v_measured == 0 || v_measured > (FRONT_FAN_CTRL_MAX_VALUE_MV + ADC_EPSILON_MV)){`

File: projects/power_distribution/src/pd_fan_ctrl.c (hunk 1 above)
Comment (JarvisWeng) [reply projects/power_distribution/src/pd_fan_ctrl.c:60]: Oh it's if you divide 0 by 1650. It shouldn't matter but I put it in there to be explicit

File: projects/power_distribution/src/pd_fan_ctrl.c (hunk 2 above)
Comment (JarvisWeng) [reply projects/power_distribution/src/pd_fan_ctrl.c:66]: We could do that

File: projects/power_distribution/src/pd_fan_ctrl.c [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#include "pd_fan_ctrl.h"
```
Comment (AveryChiu64) [inline projects/power_distribution/src/pd_fan_ctrl.c:1]: Is there a reason this was moved to the top?

File: projects/power_distribution/test/test_pd_fan_ctrl.c [hunk 4]
```diff
@@ -201,15 +202,21 @@ void test_front_pd_fan_ctrl_pot(void) {
...
+  // set adc to open value
   adc_ret_val = ADC_MAX_VAL;
   TEST_ASSERT_EQUAL(STATUS_CODE_OK, pd_fan_ctrl_init(&s_fan_settings, true));
   delay_ms(REAR_FAN_CONTROL_REFRESH_PERIOD_MILLIS + 10);
+  TEST_ASSERT_EQUAL(FAN_MIN_I2C_WRITE, i2c_buf1[1]);
+  TEST_ASSERT_EQUAL(FAN_MIN_I2C_WRITE, i2c_buf2[1]);
+
+  // set adc to return max value
+  adc_ret_val = FRONT_FAN_CTRL_MAX_VALUE_MV;
+  delay_ms(REAR_FAN_CONTROL_REFRESH_PERIOD_MILLIS + 10);
```
Comment (AveryChiu64) [inline projects/power_distribution/test/test_pd_fan_ctrl.c:214]: I noticed that line 219 and 208 have a delay with a 10 ms tolerance. Why is this not needed in line 220?

File: projects/power_distribution/src/pd_fan_ctrl.c (hunk 3 above)
Comment (JarvisWeng) [reply projects/power_distribution/src/pd_fan_ctrl.c:1]: No I think that was just my auto formatter.

File: projects/power_distribution/test/test_pd_fan_ctrl.c (hunk 4 above)
Comment (JarvisWeng) [reply projects/power_distribution/test/test_pd_fan_ctrl.c:214]: Hm not sure. That's a @mitchellostler question. I simply used the same format again to test for 3.3V

Comment (mitchellostler) [reply projects/power_distribution/test/test_pd_fan_ctrl.c:214]: I'm pretty sure the tolerance was just added to ensure that values are available. If the tests are passing it doesn't really need a change, but might be a good idea to add it to keep it consistent

Comment (JarvisWeng) [reply projects/power_distribution/test/test_pd_fan_ctrl.c:214]: Sure I can do that

**Review by ryandancy — APPROVED** [review]
LGTM!

File: projects/power_distribution/src/pd_fan_ctrl.c [hunk 5]
```diff
@@ -56,7 +57,12 @@ static void prv_front_temp_to_fan_percent(uint16_t v_measured, uint8_t *fan_spee
...
     return;
   }
-  double ratio = s_fan_storage.ref_reading / 100.0;
+  if ((v_measured == 0) || (v_measured > (FRONT_FAN_CTRL_MAX_VALUE_MV + ADC_EPSILON_MV))) {
+    *fan_speed = 0;
+    return;
+  }
+
+  double ratio = FRONT_FAN_CTRL_MAX_VALUE_MV / 100.0;
   *fan_speed = v_measured / ratio;
```
Comment (ryandancy) [inline projects/power_distribution/src/pd_fan_ctrl.c:66]: Actually wait I think that if `v_measured > FRONT_FAN_CTRL_MAX_VALUE_MV`, but `v_measured < FRONT_FAN_CTRL_MAX_VALUE_MV + ADC_EPSILON_MV`, we could end up outputting a fan speed over 100. Could you `MIN()` the calculated fan speed with 100 to make sure that doesn't happen?

Comment (JarvisWeng) [reply projects/power_distribution/src/pd_fan_ctrl.c:66]: Sure, I'm guessing I don't have to worry about negative voltages, or swapped polarities

**Pull request merged**